### PR TITLE
Switching ColumnView from a managed Kokkos View to an unmanaged one

### PR DESCRIPTION
**Pull request by jeff-cohere**
In production, we are going to assemble our Atmosphere/Prognostics/Diagnostics types using subviews representing tracers stored in EAMxx's atmosphere driver. This means that these views (represented by the `haero::ColumnView` type) must be unmanaged Kokkos views, since the AD controls their resources.

On the other hand, we need these views to be "managed" for our unit and validation tests, so we need a way to construct "standalone" ColumnViews and Atmosphere objects. In Haero, we allow this using the following new functions:

1. `haero::ColumnView haero::testing::create_column_view(int num_levels)` - creates a `ColumnView` with memory managed by a column "pool" allocator with the given number of levels. This is not a very good implementation of a memory pool, and should only be used for testing
2. `haero::Atmosphere haero::testing::create_atmosphere(int num_levels, haero::Real pblh)` - creates an `Atmosphere` with views allocated using `create_column_view` above, with the given number of levels. The `Atmosphere` object is assigned the given planetary boundary layer height.

These "standalone" views are destroyed when `haero::testing::finalize()` is called. We provide overrides to the setup/breakdown functions used by EKAT to initialize and finalize test sessions. This requires an additional parameter (`EXCLUDE_TEST_SESSION`) to be provided to the `EkatCreateUnitTest` CMake function. I've jotted down some comments where needed.

This PR has a companion PR in the mam4xx repo that I'll create now. The two PRs must be merged in tandem. Sorry for the disruption!

### CODE REVIEW

**Review by odiazib — APPROVED** [review]

**Review by pressel — APPROVED** [review]
Thanks, Jeff. This looks good to me!

File: haero/testing.cpp [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+using Real = haero::Real;
+
+// A simple memory allocation pool for standalone ColumnViews to be used in
+// (e.g.) unit tests. A ColumnPool manages a number of ColumnViews with a fixed
+// number of vertical levels.
+class ColumnPool {
+  size_t num_levels_;          // number of vertical levels per column (fixed)
+  size_t num_cols_;            // number of allocated columns
+  std::vector<int> col_used_;  // columns that are being used already
```
Comment (mjs271) [inline haero/testing.cpp:21]: Why `int` instead of `bool`? Equivalent, but just curious if there's a clever reason.

**Review by mjs271 — APPROVED** [review]
Looks good! Compiles and tests pass on my hacked M1 setup

File: haero/testing.cpp (hunk 1 above)
Comment (jeff-cohere) [reply haero/testing.cpp:21]: Glad you asked! https://stackoverflow.com/questions/17794569/why-isnt-vectorbool-a-stl-container

Comment (mjs271) [reply haero/testing.cpp:21]: ryan-gosling-slow-clapping-q1x52yju6t9wn39v

**Review by jaelynlitz — APPROVED** [review]
LGTM

**Review by overfelt — APPROVED** [review]

**Pull request merged**